File: submissions/sm_001_aalind/week_03/day_6/assignment/pyramidPlay.js

function repeatWord(word, times) {
  var repeated = "";
  
  for(var i = 0; i < times; i++) {
    repeated += word;
  }

  return repeated;
}

function pyramid(level) {
  var row = "";

  for(i = level; i >= 1; i--) {
    row = repeatWord(".", i) + repeatWord("o.", level-i+1) + repeatWord(".", i-1);
    console.log(row);
    row = "";
  }
}

function inversePyramid(level) {
  var row = "";

  for(var i = 1; i <= level; i++) {
    row = repeatWord(".", i) + repeatWord("o.", level-i+1) + repeatWord(".", i-1);
    console.log(row);
    row = "";
  }
}

function rhombus(level) {
  pyramid(level);
  for(var i = 2; i <= level; i++) {
    row = repeatWord(".", i) + repeatWord("o.", level-i+1) + repeatWord(".", i-1);
    console.log(row);
    row = "";
  }
}

// rhombus(4);

File: submissions/sm_001_aalind/week_13/day_5/session_1/pyramid_play.py
def pyramid(level):
    step = level

    for i in range(1, level+1):
        print('.'*(step-1) + '.o'*i + '.'*step)
        step -= 1

def inverse_pyramid(level):
    step = 1

    for i in range(level, 0, -1):
        print('.'*(step-1) + '.o'*i + '.'*step)
        step += 1

def rhombus(level):
    step = level

    for i in range(1, level):
        print('.'*(step-1) + '.o'*i + '.'*step)
        step -= 1

    inverse_pyramid(level)

rhombus(4)